dependency/context.go

```diff
@@ -82,6 +83,21 @@ func (ctx *context) rawAccess(resourceName string, out interface{}) error {
 	return convert(input, out)
 }
 
+// ScopedContext returns the context.Context that corresponds to
+// dependency.Context. If the dependency.Context is aborted then the
+// context.Context is canceled.
+func ScopedContext(context Context) stdcontext.Context {
+	ctx, cancel := stdcontext.WithCancel(stdcontext.Background())
+	go func() {
+		select {
+		case <-context.Abort():
+			cancel()
+		case <-ctx.Done():
+		}
+	}()
+	return ctx
+}
```

Comment:
This is somewhat ok, but means that every time you need a context.Context you'd be creating a new one.

given that the documentation for `dependency.Context` is that it only exists for the lifetime of `StartFunc`, it very much feels like we should be trying to replace its `Abort()` functionality.

I'm still not sure whether it would be better to just embed context.Context so that everything can move over to ctx.Done() instead of context.Abort.

I know we still need a way to pass in Get(), but I think other than naming conflict, it is really doing 2 different things, and it might be better if those were 2 different objects.

Reply:
> I know we still need a way to pass in Get(), but I think other than naming conflict, it is really doing 2 different things, and it might be better if those were 2 different objects.

Agreed. https://github.com/juju/worker/pull/31